File: apollo-runtime/src/jvmMain/java/com/apollographql/apollo/subscription/OnSubscriptionManagerStateChangeListener.java
package com.apollographql.apollo.subscription;

/**
 * Represents a callback for subscription manager state changes.
 */
public interface OnSubscriptionManagerStateChangeListener {
  /**
   * Called when subscription manager state changed.
   *
   * @param fromState previous subscription manager state
   * @param toState   new subscription manager state
   */
  void onStateChange(SubscriptionManagerState fromState, SubscriptionManagerState toState);
}

File: apollo-runtime/src/main/java/com/apollographql/apollo/subscription/OnSubscriptionManagerStateChangeListener.kt
package com.apollographql.apollo.subscription

/**
 * Represents a callback for subscription manager state changes.
 */
interface OnSubscriptionManagerStateChangeListener {
  /**
   * Called when subscription manager state changed.
   *
   * @param fromState previous subscription manager state
   * @param toState   new subscription manager state
   */
  fun onStateChange(fromState: SubscriptionManagerState?, toState: SubscriptionManagerState?)
}